blocks/rich-text/index.js

```diff
@@ -418,6 +436,50 @@ export class RichText extends Component {
 		};
 	}
 
+	setInsertPosition() {
+		const container = this.getContainerNode();
+		const containerStyle = window.getComputedStyle( container );
+		const marginLeft = get( containerStyle, 'margin-left', 0 );
+		const containerPosition = container.getBoundingClientRect();
+		const rect = getRectangleFromRange( this.editor.selection.getRng() );
+		const insertPosition = {
+			top: rect.top - containerPosition.top,
+			left: rect.right - containerPosition.left,
+			marginLeft,
+			height: rect.height,
+		};
+
+		this.setState( { insertPosition } );
+	}
+
+	insertInlineBlock() {
+		const { inlineBlockForInsert, completeInlineInsert } = this.props;
+
+		if ( inlineBlockForInsert.type === 'image' ) {
```

Comment:
Any way to avoid inline block specific logic here?

Reply:
I think I will try moving all of the inline block logic out of the RichText `index.js` and into a separate `InlineBlocks` component that RichText will render. I'll see if I can move some of this logic to the registered inline block as well.

Reply:
I've now moved all of the inline block logic into a new `InlineBlocks` component in https://github.com/WordPress/gutenberg/pull/5794/commits/d4229ef3135d85d0079d1c6f4481deb1192ec6f0.